usermods/Bar_Timer/usermod_timer_bar.h

```diff
@@ -0,0 +1,210 @@
+ * 
+ * 1) (device topic)/timerbar/status with payload "START" -> Feedback to know the timer actually started
+ * 2) (device topic)/timerbar/status with payload "END" -> To let know the timer has ended
+ * 
+ * Everytime the timer ends, the strip returns in the state before the timer start command (effect, palette...)
+*/
+
+
+
+
+class UsermodTimerBar : public Usermod {
+
+    private:
+
+        uint8_t timer_status;
+        uint16_t number_of_leds, led_turn_off = 0;
+        uint32_t millis_wait_time; 
+        uint16_t start_time, end_time;
+        Segment main;
+        byte settings[3];
+        int helper  = 0;
+
+
+        void stop_timer() {
+
+          led_turn_off = 0;
+          
+          //Restore previous strip settings
+          effectSpeed = settings[1];
+          effectPalette = settings[2];
+          strip.setMode(0, effectCurrent);
+
+          publishMqtt("END", false);
+
+          timer_status = STOPPED;
+        }
+
+
+        void check_timer() {
+            
+          end_time = millis();
+
+          if ((end_time - start_time) >= millis_wait_time) {
+
+            led_turn_off++;
+            start_time = end_time;
+
+            if (led_turn_off == number_of_leds) {
+
+              stop_timer();
+            }
+          }         
+        }        
+
+
+
+    public:
+
+        void setup() {
+
+          timer_status = STOPPED;
+          led_turn_off = 0;
+        }
+    
+        void loop (){
+
+          if (timer_status == RUNNING) {
+
+            helper = 1;
+            strip.trigger();
+            strip.show();
+
+            for (int y = number_of_leds - 1; y >= number_of_leds - led_turn_off; y--) {
+
+              strip.setPixelColor(y, 0, 0, 0, 0);              
+            }
+
+            check_timer();
+          }
+          helper = 0;
+        }
+
+
+        /**
+         * This function is executed before every strip.show(). This timer visualization uses the solid effect. When leds are turned off, the main
+         * WLED software try to restore the color picked from the web UI, so time by time all the turned off leds light up again and then turned off
+         * in the loop function of this usermod. So using this function we can set again the led off before the show() function
+         * 
+         */
+        void handleOverlayDraw()
+        {
+          if (helper == 0 && timer_status == RUNNING) {
+          
+            for (int y = number_of_leds - 1; y >= number_of_leds - 1 - led_turn_off; y--) {
+            
+             strip.setPixelColor(y, 0, 0, 0, 0);              
+            }
+          }
+        }
+
+
+
+
+    /**
+     * handling of MQTT message
+     * topic only contains stripped topic (part after /wled/MAC)
+     */
+        bool onMqttMessage(char* topic, char* payload) {
+
+          // check if we received a command
+
+          if (strlen(topic) == 9 && strncmp_P(topic, PSTR("/timerbar"), 9) == 0) {
+
+            if (strcmp("STOP", payload) == 0){
```

Comment:
Please be consistent, use `..._P()` functions and strings in PROGMEM if possible.